tests/gordo_components/workflow/test_config_elements.py

```diff
@@ -182,3 +182,41 @@ def test_from_config(self):
                 },
             },
         )
+
+    def test_invalid_model(self):
+        """
+        Test invalid model with step instead of steps
+        """
+        element_str = """
+            name: ct-23-0001-machine
+            data_provider:
+              threads: 10
+            dataset:
+              tags: [GRA-TE  -23-0733.PV, GRA-TT  -23-0719.PV, GRA-YE  -23-0751X.PV]
+              target_tag_list: [GRA-TE -123-456]
+              train_start_date: 2018-01-01T09:00:30Z
+              train_end_date: 2018-01-02T09:00:30Z
+            model:
+              sklearn.pipeline.Pipeline:
+                step:
+                  - sklearn.preprocessing.data.MinMaxScaler
+                  - gordo_components.model.models.KerasAutoEncoder:
+                      kind: feedforward_hourglass
+            evaluation:
+                scoring_scaler: Null
+            metadata:
+              id: special-id
+        """
+        yaml.FullLoader.add_constructor(
```

Comment:
Maybe you could use `workflow_generator.get_dict_from_yaml` here?

Reply:
@milesgranger Then I belive the function in `workflow_generator.get_dict_from_yaml` has to be rewritten. It seems that it only will work for `io.StringIO`, as a string will fail on `if hasattr(config_file, "read")`. So it should probably be rewritten anyway since it is supposed to accept `str`.